tensorflow/python/data/ops/dataset_ops.py

```diff
@@ -973,6 +973,23 @@ def skip(self, count):
   def shard(self, num_shards, index):
     """Creates a `Dataset` that includes only 1/`num_shards` of this dataset.
 
+    For example:
+    
+    ```python
+    # Create a Dataset with 60 elements.
+    A = tf.data.Dataset.range(60) # ==> [0, 1, 2, 3, ..., 57, 58, 59]
+
+    # Create 3 Datasets, each with 20 elements from Dataset A.
+    B = A.shard(num_shards=3, index=0) # ==> [0, 3, 6, 9, ..., 51, 54, 57]
+    C = A.shard(num_shards=3, index=1) # ==> [1, 4, 7, 10, ..., 52, 55, 58]
+    D = A.shard(num_shards=3, index=2) # ==> [2, 5, 8, 11, ..., 53, 56, 59]
+
+    # There is no overlap between Datasets B, C and D.
+    ```
+    
+    `shard` is a deterministic operator; the returned Dataset depends only
```

Comment:
We can make this more precise: "The dataset produced by `A.shard(n, i)` will contain all elements of `A` where their index mod `n` equals `i`."

Can we move this sentence to replace "For example:" above?